### PR TITLE
Add `objectscript.openClassContracted` setting

### LINKED ISSUE #876: Optionally fold all ranges when class is opened
> In studio there was the option to "Open class in contracted view"
In vs code, for each class I open, I have to click in the class and ctrl+k ctrl+1 to get the same effect.
Can it be made a setting and auto-applied?

This is possible using the `editor.foldAll` built  in command, which  s what gets called when you use the `ctrl-k-0` keybinding.

*Moved from https://github.com/intersystems/language-server/issues/218*

### PR DESCRIPTION
**Pull request by isc-bsaviano**
This PR fixes #876 

When `objectscript.openClassContracted` is `true`, automatically collapse all folding ranges when a class is opened for the first time.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -738,6 +738,9 @@ export async function activate(context: vscode.ExtensionContext): Promise<any> {
     }
   }
 
+  // The URI's of all classes that have been opened. Used when objectscript.openClassContracted is true.
```
Comment (gjsjohnmurray) [inline src/extension.ts:741]: Nitpick - remove apostrophe:

```suggestion
  // The URIs of all classes that have been opened. Used when objectscript.openClassContracted is true.
```

**Review by gjsjohnmurray — COMMENTED** [review]
What happens when reopening after hot exit? My guess is the open classes all get collapsed.

**Comment by isc-bsaviano** [comment]
@gjsjohnmurray I would expect that open classes would be collapsed when they become the active editor but I haven't tested it because I don't have #881 in that branch.

**Comment by gjsjohnmurray** [comment]
@isc-bsaviano once #881 merges and you update your branch I think it's worth investigating. Maybe move the `openedClasses` array into `workspaceState` so it survives reloads.

**Comment by isc-bsaviano** [comment]
@gjsjohnmurray If we persist `openClasses` then I think we should also remove classes from it when they've been closed.

**Comment by gjsjohnmurray** [comment]
Should perhaps do that even if we don't persist.

**Comment by isc-bsaviano** [comment]
@gjsjohnmurray I confirmed that when restoring from a hot exit, the class is auto-folded again even though it was open and some ranges were left unfolded. I will store `openClasses` in `workspaceState` so this doesn't happen.

**Review by gjsjohnmurray — COMMENTED** [review]
May be less of an overhead if you fetched `openedClasses` from workspace storage at the start of `activate`, updated the local array on the callbacks, then stored it during `deactivate`.

File: src/extension.ts [hunk 2]
```diff
@@ -969,7 +974,6 @@ export async function activate(context: vscode.ExtensionContext): Promise<any> {
       )
     ),
     vscode.window.onDidChangeActiveTextEditor((editor: vscode.TextEditor) => {
```
Comment (gjsjohnmurray) [inline src/extension.ts:976]: Suggest you use a `const` for `editor.document.uri.toString()` in place of the two occurrences below.

File: src/extension.ts [hunk 3]
```diff
@@ -978,15 +982,12 @@ export async function activate(context: vscode.ExtensionContext): Promise<any> {
       ) {
         vscode.commands.executeCommand("editor.foldAll");
         openedClasses.push(editor.document.uri.toString());
-        workspaceState.update("openedClasses", openedClasses);
       }
     }),
     vscode.workspace.onDidCloseTextDocument((doc: vscode.TextDocument) => {
```
Comment (gjsjohnmurray) [inline src/extension.ts:987]: Suggest you use a `const` for `doc.uri.toString()` in place of the two occurrences below.

**Review by gjsjohnmurray — CHANGES_REQUESTED** [review]
A couple of improvements suggested.

**Review by gjsjohnmurray — APPROVED** [review]

**Pull request merged**